tests/fs-tests-runner/src/lib.rs

```diff
@@ -60,7 +60,7 @@ pub struct TestLoaderParams<'a> {
     pub test_name: &'a str,
 }
 
-pub type TestLoader = Box<dyn Fn(TestLoaderParams) -> Option<Trial>>;
+pub type TestLoader = Box<dyn Fn(TestLoaderParams, &mut dyn FnMut(Trial))>;
```

Comment:
It would be nicer to return an iterator of `Trial` instead of operating on mutable state.